cmd/spire-server/cli/entry/show.go

```diff
@@ -154,52 +178,42 @@ func (c *showCommand) fetchByEntryID(ctx context.Context, id string, client entr
 	return entry, nil
 }
 
-// filterByFederatedWith evicts any value from the given entries slice that does
-// not contain at least one of the federated trust domains specified in the
-// federatesWith slice.
-func (c *showCommand) filterByFederatedWith(entries []*types.Entry) []*types.Entry {
-	// Build map for quick search
-	var federatedIDs map[string]bool
-	if len(c.federatesWith) > 0 {
-		federatedIDs = make(map[string]bool)
-		for _, federatesWith := range c.federatesWith {
-			federatedIDs[federatesWith] = true
-		}
-	}
+func printEntries(entries []*types.Entry, env *common_cli.Env) {
+	msg := fmt.Sprintf("Found %v ", len(entries))
+	msg = util.Pluralizer(msg, "entry", "entries", len(entries))
 
-	// Filter slice in place
-	idx := 0
+	env.Println(msg)
 	for _, e := range entries {
-		if keepEntry(e, federatedIDs) {
-			entries[idx] = e
-			idx++
-		}
+		printEntry(e, env.Printf)
 	}
-
-	return entries[:idx]
 }
 
-func keepEntry(e *types.Entry, federatedIDs map[string]bool) bool {
-	// If FederatesWith was specified, discard entries that don't match
-	if federatedIDs == nil {
-		return true
-	}
-
-	for _, federatesWith := range e.FederatesWith {
-		if federatedIDs[federatesWith] {
-			return true
-		}
+func parseToSelectorMatch(match string) (types.SelectorMatch_MatchBehavior, error) {
+	switch match {
+	case "exact":
+		return types.SelectorMatch_MATCH_EXACT, nil
+	case "any":
+		return types.SelectorMatch_MATCH_ANY, nil
+	case "superset":
+		return types.SelectorMatch_MATCH_SUPERSET, nil
+	case "subset":
+		return types.SelectorMatch_MATCH_SUBSET, nil
+	default:
+		return types.SelectorMatch_MATCH_SUPERSET, errors.New("unsupported match behavior")
```

Comment:
It could be good to give the match name back to the user with the error so it's easier to spot typos etc

```suggestion
		return types.SelectorMatch_MATCH_SUPERSET, fmt.Errorf("match behavior %q unknown", match)
```

Reply:
done